rspace/src/main/scala/coop/rchain/rspace/SpaceMatcher.scala

```diff
@@ -43,33 +44,28 @@ private[rspace] trait SpaceMatcher[F[_], C, P, E, A, R, K] extends ISpace[F, C,
     * along with the remaining unmatched data. If an illegal state is reached
     * during searching for a match we short circuit and return the state.
     */
-  @tailrec
   private[rspace] final def findMatchingDataCandidate(
       channel: C,
       data: Seq[(Datum[A], Int)],
       pattern: P,
       prefix: Seq[(Datum[A], Int)]
-  )(implicit m: Match[P, E, A, R]): Either[E, Option[(DataCandidate[C, R], Seq[(Datum[A], Int)])]] =
+  )(
+      implicit m: Match[F, P, A, R]
+  ): F[Option[(DataCandidate[C, R], Seq[(Datum[A], Int)])]] =
```

Comment:
pls introduce a type alias for the repeated type signature

Reply:
👍